### ISSUE TITLE
Failure to generate dependency snapshot

### ISSUE THREAD

**Issue by atul-exabeam** [advanced-security/maven-dependency-submission-action#11]
The GH action seems to work on few of my repos but gives the error below on a couple of them.  Any thoughts on why this is happening would be appreciated. Thanks

```
Error: TypeError: Cannot read properties of undefined (reading 'forEach')
Error: Error: Could not generate a snapshot of the dependencies; Cannot read properties of undefined (reading 'forEach')
Error: Failed to generate a dependency snapshot, check logs for more details, Error: Could not generate a snapshot of the dependencies; Cannot read properties of undefined (reading 'forEach')
```

**Comment by felickz** [comment]
Hi there @atul-exabeam, looks like the tooling is struggling to find a direct dependency of your root artifact.  Could you provide some details around your package manifest or a simplified/redacted one you could share?

**Comment by atul-exabeam** [comment]
@felickz - One of the repos it is failing on the root pom for our Java projects. Is maven-dependency-submission-action supposed to work on root pom. The other project it does not work on is one with multiple module. Will try to get more details on it soon.

**Comment by felickz** [comment]
For the multiple module project, you can test out early support for Reactor in your action by using the `multi-module` feature branch:
```yaml
      uses: advanced-security/maven-dependency-submission-action@multi-module
```

**Comment by atul-exabeam** [comment]
@felickz  I tried it one of our multi-module repos and it works perfectly. For another one, I get the following error

Error: RangeError: Maximum call stack size exceeded
Error: Error: Could not generate a snapshot of the dependencies; Maximum call stack size exceeded
Error: Failed to generate a dependency snapshot, check logs for more details, Error: Could not generate a snapshot of the dependencies; Maximum call stack size exceeded

**Comment by atul-exabeam** [comment]
@felickz  See attached screenshot
![Screen Shot 2022-11-29 at 14 34 55](https://user-images.githubusercontent.com/104833269/204663369-618b1653-a30a-4d7d-9d73-33bebb307ea5.png)

**Comment by peter-murray** [comment]
This has been resolved in version `v3.0.0` with the added support of the full Maven coordinate system in the PackageURLs that the action builds in to the dependency snapshot
